Examples/MulticurveBootstrapping/MulticurveBootstrapping.cpp

```diff
@@ -731,10 +728,6 @@ int main(int, char* []) {
                    "5-years swap mispriced by "
                    << io::rate(std::fabs(fairRate-s5yRate->value())));
 
-
-        forecastingTermStructure.linkTo(euribor6MTermStructure);
-        discountingTermStructure.linkTo(euribor6MTermStructure);
-
```

Comment:
No, this breaks the logic of the example, which is to show that instruments recalculate consistently when different curves are used for forecasting and discounting.  I'm not sure how this can cause a cycle, but the fix will be different.

Reply:
You're right, I missed how it was actually different term structures being linked to throughout the example. Also `Handle<T>::Link::linkTo` doesn't relink the handle if it's equal to the one already linked, so linking to the same handle more than once should not be a problem.

I still don't know what is causing the indirect leaks in this example. I tried linking to empty handle like I did in the inflation test, but that didn't work (not that I expected it would work - it was just a guess). I will need more time to figure out what is going on, unless you can think of someone else that might know more about this than me.

Reply:
I'll try to have a look next week.

Reply:
Ok, linking `discountingTermStructure ` to `euribor6MTermStructure` causes a cycle (since the latter uses the former indirectly through its helpers).  I might remove this part from the example — it is equivalent to the old single-curve calculation, but I'm not sure it's really of interest.

Reply:
Thanks for having a look at this and locating the cycle. I've pulled in your latest commits on master to my branch and kicked off a new [full test](https://github.com/sweemer/QuantLib/actions/runs/1645718728) to check that there are no more indirect leaks. Assuming that there aren't any, then what are you thoughts on the following?

1. Should we keep the AddressSanitizer in the full test matrix? Since this already has a long runtime and is only run manually, I think it's definitely worth the additional cost.
2. Can we add runtime checks for observer cycles? If there have been unsuccessful attempts at this in the past, then are there any other approaches we could try?